mdagent/utils/set_ckpt.py

```diff
@@ -0,0 +1,62 @@
+import os
+import shutil
+
+
+class SetCheckpoint:
+    def find_root_dir(self):
```

Comment:
i have a proposition: 
```
def find_project_root(stop_at="md-agent"):
    """
    Traverse up from the current_path to find the stop_at directory.

    :param current_path: The starting point for the search.
    :param stop_at: The directory name to stop the search at.
    :return: The path to the directory named stop_at, or just current_path if not in
    a directory named stop_at or any of its parents.
    """
    current_path = Path(os.getcwd())
    if current_path.name == stop_at:
        return current_path
    for parent in current_path.parents:
        if parent.name == stop_at:
            return parent
    return current_path
```
I had this so that the ckpt will always go to the root directory of md-agents, or in case we're not working inside the repo, it will just be the local directory.